File: solutions/javascript/twelve-days/1/twelve-days.js
//
// This is only a SKELETON file for the 'Twelve Days' exercise. It's been provided as a
// convenience to get you started writing code faster.
//
const gifts = [
  'and a Partridge in a Pear Tree',
  'two Turtle Doves',
  'three French Hens',
  'four Calling Birds',
  'five Gold Rings',
  'six Geese-a-Laying',
  'seven Swans-a-Swimming',
  'eight Maids-a-Milking',
  'nine Ladies Dancing',
  'ten Lords-a-Leaping',
  'eleven Pipers Piping',
  'twelve Drummers Drumming'
];

const ordinals = [
  'first',
  'second',
  'third',
  'fourth',
  'fifth',
  'sixth',
  'seventh',
  'eighth',
  'ninth',
  'tenth',
  'eleventh',
  'twelfth'
];

export const recite = (start, end = start) => {
  const verses = [];
  for (; start <= end; start++) {
    const current_gifts = gifts.slice(0, start).reverse();
    if (start == 1)
      current_gifts[0] = current_gifts[0].slice(4);
    const verse = `On the ${ordinals[start-1]} day of Christmas my true love gave to me: ${current_gifts.join(', ')}.\n`;
    verses.push(verse);
  }
  return verses.join('\n');
};

File: solutions/python/twelve-days/1/twelve_days.py
def recite(start_verse, end_verse):
    verses = [
        ("first", "a Partridge in a Pear Tree."),
        ("second", "two Turtle Doves, and "),
        ("third", "three French Hens, "),
        ("fourth", "four Calling Birds, "),
        ("fifth", "five Gold Rings, "),
        ("sixth", "six Geese-a-Laying, "),
        ("seventh", "seven Swans-a-Swimming, "),
        ("eighth", "eight Maids-a-Milking, "),
        ("ninth", "nine Ladies Dancing, "),
        ("tenth", "ten Lords-a-Leaping, "),
        ("eleventh", "eleven Pipers Piping, "),
        ("twelfth", "twelve Drummers Drumming, ")
    ]
    result = []
    for i in range(start_verse - 1, end_verse):
        current = f"On the {verses[i][0]} day of Christmas my true love gave to me: "
        for j in range(i, -1, -1):
            current += verses[j][1]
        result.append(current)
    return result
